### PR TITLE
Feature/3713: QA & Cert: Test Summary & Linearity Data Validations

### PR DESCRIPTION
**Pull request by subrataroyCVP**

### CODE REVIEW

File: src/linearity-injection-workspace/linearity-injection-checks.service.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    );
+    records.forEach(record => {
+      if (
+        record.injectionDate.toDateString() ===
+          linearityInjection.injectionDate.toDateString() &&
+        record.injectionHour === linearityInjection.injectionHour &&
+        record.injectionMinute === linearityInjection.injectionMinute
+      ) {
+        // LINEAR-33 Duplicate Linearity Injection (Result A)
+        error = `Another Linearity Injection record already exists with the same injectionDate [${linearityInjection.injectionDate}], injectionHour [${linearityInjection.injectionHour}], injectionMinute [${linearityInjection.injectionMinute}].`;
```
Comment (fanwari1) [inline src/linearity-injection-workspace/linearity-injection-checks.service.ts]: Though I think this error is more helpful it's not the error message that's on the spec sheet. Did you check with vishnu before changing the text for (result A)?

Comment (subrataroyCVP) [reply src/linearity-injection-workspace/linearity-injection-checks.service.ts]: I did not made any change to error test. ⬇

image

Comment (fanwari1) [reply src/linearity-injection-workspace/linearity-injection-checks.service.ts]: Ah I was looking at the wrong one. My mistake.

File: src/linearity-summary-workspace/linearity-summary-checks.service.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    this.throwIfErrors(errorList, isImport);
+    this.logger.info('Completed Linearity Summary Checks');
+    return errorList;
+  }
+
+  private async duplicateTestCheck(
+    testSumId: string,
+    linearitySummary: LinearitySummaryBaseDTO | LinearitySummaryImportDTO,
+    _isImport: boolean = false,
```
Comment (fanwari1) [inline src/linearity-summary-workspace/linearity-summary-checks.service.ts:53]: can we remove the _ from _isImport so its consistent with the others

**Review by fanwari1 — APPROVED** [review]

File: src/linearity-summary-workspace/linearity-summary-checks.service.ts (hunk 2 above)
Comment (subrataroyCVP) [reply src/linearity-summary-workspace/linearity-summary-checks.service.ts:53]: That is intertidally added to get rid of the linting error.

**Pull request merged**